ibis/common/validators.py

```diff
@@ -30,13 +28,19 @@ class Optional(Validator):
         Value to return with in case of a missing argument.
     """
 
-    __slots__ = ('validator', 'default')
+    __slots__ = (
+        'validator',
+        'default',
+    )
 
     def __init__(self, validator, default=None):
         self.validator = validator
         self.default = default
 
-    def __call__(self, arg, **kwargs):
+    def __repr__(self):
+        return f"Optional({self.validator!r}, {self.default!r})"
+
+    def validate(self, arg, **kwargs):
```

Comment:
Why can't we preserve `__call__`?

Reply:
It's an experimental PR to see the performance gain if any. 

Preserving `__call__` would require a decent amount of signature tracking, see the `toolz.curry` implementation: https://github.com/pytoolz/toolz/blob/f3ac8211f510edff77c1479061d99bb2dc8ab878/toolz/functoolz.py#L170-L377